task_lib.c

```diff
@@ -2564,6 +2590,49 @@ task_create(struct instance *i, int type, int fd, struct worker *w, void *close_
 	return t;
 } // task_create
 
+static ssize_t
+SSL_writev (SSL *ssl, const struct iovec *vector, int count)
+{
+  char *buffer;
+  register char *bp;
+  size_t bytes, to_copy;
+  int i;
+
+  /* Find the total number of bytes to be written.  */
+  bytes = 0;
+  for (i = 0; i < count; ++i)
+    bytes += vector[i].iov_len;
+
+  /* Allocate a temporary buffer to hold the data.  */
+  buffer = (char *) alloca (bytes);
```

Comment:
I was also wondering if that could mostly be done with a stack allocated buffer.  As in, define a local char [] array of, say, 64K bytes in size.  eg. `char buffer[65536]'`.  If the length of the iovecs (`bytes`) is <= 65536, use the stack buffer (`bp = buffer;`)  If greater than, do an `aligned_alloc` to grab enough system pages to assemble the buffer, and then release them again with a regular free when done.  eg. `bp = aligned_alloc(__page_size, (((bytes - 1) / __page_size) + 1) * __page_size);`

Use the `char *bp` variable to either point at the stack buffer, or the aligned_alloc buffer, and then just prior to exiting the function, check if the bp == buffer.  If not, do a free on it.

Just make sure to reference the buffer by the local stack variable name, and not the direct stack buffer character array, and you should be golden.  Within the loop where you're building the buffer, use a different `char *` variable to track the location of where you're copying the bytes into mid-buffer, and preserve `bp` to be the base of the actual buffer, whether on the stack, or alloced.